packages/datasource-sparql/lib/datasources/SparqlDatasource.js

```diff
@@ -139,39 +139,69 @@ class SparqlDatasource extends Datasource {
 
   // Creates a SPARQL pattern for the given triple pattern
   _createQuadPattern(quad) {
-    var query = ['{'], literalMatch;
+    var query = ['{'];
 
     // Encapsulate in graph if we are not querying the default graph
-    if (quad.graph !== '') {
+    if (quad.graph !== '' && !N3.DataFactory.defaultGraph().equals(quad.graph)) {
```

Comment:
`quad.graph` should not be able to be either a string or a term, so one of these checks should go.

Reply:
Correct, it cannot be a string